src/fondant/cli.py

```diff
@@ -840,47 +872,45 @@ def pipeline_from_string(string_ref: str) -> Pipeline:  # noqa: PLR0912
             if not _called_with_wrong_args(attr):
                 raise
 
-            msg = f"The factory {pipeline_str} in module {module.__name__} could not be called with the specified arguments."
-            raise PipelineImportError(
+            msg = f"The factory {dataset_str} in module {module.__name__} could not be called with the specified arguments."
+            raise DatasetImportError(
                 msg,
             ) from e
     else:
         app = attr
 
-    if isinstance(app, Pipeline):
+    if isinstance(app, Dataset):
         return app
 
-    msg = f"A valid Fondant pipeline was not obtained from '{module.__name__}:{pipeline_str}'."
-    raise PipelineImportError(
+    msg = f"A valid Fondant workspace was not obtained from '{module.__name__}:{dataset_str}'."
+    raise DatasetImportError(
         msg,
     )
 
 
-def pipeline_from_module(module_str: str) -> Pipeline:
-    """Try to import a pipeline from a string otherwise raise an ImportFromStringError."""
-    from fondant.pipeline import Pipeline
+def dataset_from_module(module_str: str) -> Dataset:
+    """Try to import a dataset from a string otherwise raise an ImportFromStringError."""
+    from fondant.dataset import Dataset
 
     module = get_module(module_str)
 
-    pipeline_instances = [
-        obj for obj in module.__dict__.values() if isinstance(obj, Pipeline)
+    dataset_instances = [
+        obj for obj in module.__dict__.values() if isinstance(obj, Dataset)
     ]
 
-    if not pipeline_instances:
-        msg = f"No pipeline found in module {module_str}"
-        raise PipelineImportError(msg)
+    if not dataset_instances:
+        msg = f"No dataset found in module {module_str}"
+        raise DatasetImportError(msg)
 
-    if len(pipeline_instances) > 1:
+    # Skip this one and choose the first dataset instance?
+    if len(dataset_instances) > 1:
         msg = (
-            f"Found multiple instantiated pipelines in {module_str}. Only one pipeline "
-            f"can be present"
+            f"Found multiple instantiated datasets in {module_str}. Use the first dataset to start "
+            f"the execution."
```

Comment:
```suggestion
            f"Found multiple instantiated datasets in {module_str}. Using the first dataset to start "
            f"the execution."
```

Shouldn't we use the last dataset? Or should it not matter? And if it doesn't matter, should we even log this?

Reply:
I've checked it again. We don't need it since we don't create a deepcopy of the graph. Every dataset will contain the same computational graph after applying a new operations. We are not able to create a deepcopy since the graph contains some mappingproxy objects. Shouldn't be a problem for now, if we just allow a single dataset object in a module. 

For the case we allow something like this: 

```python
dataset_1 = Dataset.read(...)
dataset_2 = dataset_1.apply(...)
dataset_3 = dataset_2.apply(...)
```
we will run into issues. For now we could add a validation that every dataset of a module contains the same graph, wdyt?